File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        maxi=0
        mini=float('inf')
        for i in range(0,len(prices)):
            if prices[i]<mini:
                mini=prices[i]
            else:
                maxi=max(maxi,prices[i]-mini)
        return maxi
            
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini = Integer.MAX_VALUE;
        int Profit = 0;

        for(int i = 0; i < prices.length; i++){
            mini = Math.min(mini , prices[i]);
            int diff = prices[i] - mini;
            Profit = Math.max(Profit , diff);
            
        }
        
        return Profit;
    }
}